### ISSUE TITLE
Color surface clear for X1R5G5B5 is untested

### ISSUE THREAD

**Issue by JayFoxRox** [xqemu/xqemu#213]
The respective code is here:

https://github.com/xqemu/xqemu/blob/960698a24ed7cd69fe2e452a28a3bb64e395a0fe/hw/xbox/nv2a/nv2a_pgraph.c#L2325-L2331

You might have to use a tool like apitrace to verify the correct color, but for other games the color might be visible in certain spots. If it's wrong, it is probably just a wrong order of the channels.

**Affects:**

* Pro Cast Sports Fishing
* Test Drive: Off-Road - Wide Open
* The Sims 2
* Yager

Someone has to run these games and make sure the clear color is correct, if it works, remove the `assert`.

**Comment by JayFoxRox** [comment]
I'll be confirming this myself in the next days; currently waiting for https://github.com/XboxDev/nxdk/pull/141.

There'll probably be a follow-up issue about `O1R5G5B5` and `Z1R5G5B5` differences, which will be relevant when we start with a surface cache or if we want to optimize our `glReadPixels`.

**Comment by JayFoxRox** [comment]
Due to the long delay of the mentioned PR I did not test this yet. This issue is still up-for-grabs.
I recommend to test nxdk behaviour on hardware, and once that is confirmed to work, compare results on XQEMU.

*(Ideally using nv2a-trace or some other screenshot utility to confirm correctness at bit-level; but we can fix that later)*

**Comment by Triticum0** [comment]
@JayFoxRox This as been tested to match real hardware by  Ernegien back in March with this [testbed](https://github.com/Ernegien/nxdk/tree/test/texture-formats) Could you give me more info on O1R5G5B5 and Z1R5G5B5 difference so a follow up pr can be done on this colour surface clear.

**Comment by JayFoxRox** [comment]
> @JayFoxRox This as been tested to match real hardware by Ernegien back in March with this [testbed](https://github.com/Ernegien/nxdk/tree/test/texture-formats)

I don't think so, because the test runs in 32bpp mode: https://github.com/Ernegien/nxdk/blob/test/texture-formats/samples/texture-formats/main.c#L149
Textures are inputs; surfaces are outputs. The linked testbed experiments with textures, not surfaces.

> Could you give me more info on O1R5G5B5 and Z1R5G5B5 difference so a follow up pr can be done on this colour surface clear.

O1 probably stands for "One, 1-bit" whereas Z1 is "Zero, 1-bit". So, to my knowledge, this will fill the alpha bit with 0 or 1. I don't think it's possible to have alpha output with 16 bit modes (at least not 1555?).

**Comment by Triticum0** [comment]
@JayFoxRox Thanks for the info, but I make a mistake when linking you to the testbed. I thought the code was in the samples folder, but I didn't double-check, so I went back to look at the issue, and the test code was placed they when usually placed on the pull, my mistake so this is was the code:

```#include <hal/video.h>
#include <hal/xbox.h>
#include <pbkit/pbkit.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>
#include <windows.h>

int main(void)
{
    int       width = 640, height = 480;
    DWORD     color = 0b10100100010000100001000010000100;

    // NOTE: must also modify the following line in nxdk/lib/pbkit/pbkit.c
    // static unsigned int pb_ColorFmt = NV097_SET_SURFACE_FORMAT_COLOR_LE_X1R5G5B5_Z1R5G5B5;	// ORIG: NV097_SET_SURFACE_FORMAT_COLOR_LE_A8R8G8B8;
    XVideoSetMode(width, height, 15, REFRESH_DEFAULT);

    pb_init();
    while(1) {
        pb_wait_for_vbl();
        pb_reset();
        pb_target_back_buffer();
        pb_fill(0, 0, width, height, color);
        while (pb_busy() && pb_finished()) { }
        color = color >> 1;
        Sleep(5000);
    }
}
 ```
The testbed I gave you was to fix missing colour formats for #33 and #34 as well as  0x17, 0x17 and 0x3b, which hasn't been implemented.I It has been a couple of months since I have read the code, so I was misremembering. Sorry for that.
